apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/store/dot-palette.store.ts

```diff
@@ -0,0 +1,229 @@
+
+        this.setLoading();
+
+        this.filter$.pipe(take(1)).subscribe((value: string) => {
+            filter = value;
+        });
+
+        if (this.isFormContentType) {
+            this.paginationService.setExtraParams('filter', filter);
+
+            this.paginationService
+                .getWithOffset((event && event.first) || 0)
+                .pipe(take(1))
+                .subscribe((data: DotCMSContentlet[] | DotCMSContentType[]) => {
+                    data.forEach((item) => (item.contentType = item.variable = 'FORM'));
+                    this.setLoaded();
+                    this.setContentlets(data);
+                    this.setTotalRecords(this.paginationService.totalRecords);
+                });
+        } else {
+            let langId: string;
+            this.languageId$.pipe(take(1)).subscribe((value: string) => {
+                langId = value;
+            });
+
+            this.paginatorESService
+                .get({
+                    itemsPerPage: this.itemsPerPage,
+                    lang: langId || '1',
+                    filter: filter || '',
+                    offset: (event && event.first.toString()) || '0',
+                    query: `+contentType: ${this.contentTypeVarName}`
+                })
+                .pipe(take(1))
+                .subscribe((response: ESContent) => {
+                    this.setLoaded();
+                    this.setTotalRecords(response.resultsSize);
+                    this.setContentlets(response.jsonObjectView.contentlets);
+                });
+        }
+    }
+
+    // SELECTORS
+    private readonly contentlets$ = this.select(({ contentlets }) => {
+        return contentlets;
+    });
+
+    private readonly contentTypes$ = this.select(({ contentTypes }) => {
+        return contentTypes;
+    });
+
+    private readonly filter$ = this.select(({ filter }) => {
+        return filter;
+    });
+
+    private readonly totalRecords$ = this.select(({ totalRecords }) => {
+        return totalRecords;
+    });
+
+    private readonly languageId$ = this.select(({ languageId }) => {
+        return languageId;
+    });
+
+    private readonly viewContentlet$ = this.select(({ viewContentlet }) => {
+        return viewContentlet;
+    });
+
+    private readonly loading$: Observable<boolean> = this.select(
+        (state) => state.callState === LoadingState.LOADING
+    );
+
+    private isFormContentType: boolean;
+    private itemsPerPage = 25;
+    private contentTypeVarName: string;
+
+    readonly vm$ = this.select(
+        this.contentlets$,
+        this.contentTypes$,
+        this.filter$,
+        this.totalRecords$,
+        this.viewContentlet$,
+        this.loading$,
+        (contentlets, contentTypes, filter, totalRecords, viewContentlet, loading) => ({
+            contentlets,
+            contentTypes,
+            filter,
+            totalRecords,
+            viewContentlet,
+            loading
+        })
+    );
```

Comment:
Remove all the selectors that are not being used and just use the full state object if you want to skip lang id cool, but I would just send it.